src/vm/runners/cairo_runner.rs

```diff
@@ -516,20 +517,32 @@ impl CairoRunner {
     pub fn run_until_pc(
         &mut self,
         address: Relocatable,
+        run_resources: &mut Option<RunResources>,
         vm: &mut VirtualMachine,
         hint_processor: &mut dyn HintProcessor,
     ) -> Result<(), VirtualMachineError> {
         let references = self.get_reference_list();
         let hint_data_dictionary = self.get_hint_data_dictionary(&references, hint_processor)?;
+
         #[cfg(feature = "hooks")]
         vm.execute_before_first_step(self, &hint_data_dictionary)?;
-        while vm.run_context.pc != address {
+
+        while vm.run_context.pc != address
+            && !if let Some(r) = run_resources.as_ref() {
+                r.consumed()
+            } else {
+                false
+            }
+        {
```

Comment:
This looks confusing. Can you find a way to make it more straightforward?

Reply:
i think i can pull out a solution by calling a function that engulfs that behavior.